### PR TITLE
Adding support for AsyncAws

### LINKED ISSUE #638: Support for AsyncAws
AsyncAws is a light weight version of the official AWS sdk. Using the SimpleS3 client will solve issues like #634. 

Will a PR to add support for AsyncAws be accepted?

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #638

### CODE REVIEW

File: tests/Gaufrette/Functional/Adapter/AsyncAwsS3Test.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            'accessKeyId' => $key,
+            'accessKeySecret' => $secret,
+        ]);
+
+        $this->createFilesystem(['create' => true]);
+    }
+
+    protected function tearDown()
+    {
+        // TODO delete bucket
```
[the remark this answers is not in the document]
Comment (Nyholm) [reply tests/Gaufrette/Functional/Adapter/AsyncAwsS3Test.php:39]: True, I added that in the S3 client after I made this PR. 

I should update this.

**Comment by Nyholm** [comment]
Friendly ping for @nicolasmure and @nm2107 

Do you need help maintaining Gaufrette and GaufretteBundle?

**Comment by alexpozzi** [comment]
Hello @Nyholm,
@nicolasmure and @nm2107 (which are the same person :smile:) are not maintaining this library anymore so help is very welcome.
I can grant you access to this repository and the associated SF bundle if it's okay with you.

**Comment by Nyholm** [comment]
Sure. Thank you.

**Comment by alexpozzi** [comment]
Done, don't hesitate to ping me for help or assign me code reviews, I'll do my best to be reactive :wink: 

Thanks for your help!

**Comment by Nyholm** [comment]
Thank you. 

I've made sure master is green and supports PHP7.4. I've also added some tests on github actions. I removed some deprecated code. There is no point in keeping that around for long. 

I'll merge this and test on PHP8. After that I'll release 0.10.

File: src/Gaufrette/Adapter/AsyncAwsS3.php [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+    /**
+     * @param SimpleS3Client $service
+     * @param string   $bucket
+     * @param array    $options
+     * @param bool     $detectContentType
+     */
+    public function __construct(SimpleS3Client $service, $bucket, array $options = [], $detectContentType = false)
+    {
+        if (!class_exists(SimpleS3Client::class)) {
```
Comment (stof) [inline src/Gaufrette/Adapter/AsyncAwsS3.php:38]: this check is useless. If the class does not exist, it is impossible to pass an argument to `$service` due to the typehint. So this is dead code.

Comment (Nyholm) [reply src/Gaufrette/Adapter/AsyncAwsS3.php:38]: That is true. 

Thank you

**Pull request merged**